jmxtrans-output/jmxtrans-output-core/src/main/java/com/googlecode/jmxtrans/model/output/StatsDTelegrafWriter.java

```diff
@@ -0,0 +1,134 @@
+import com.googlecode.jmxtrans.model.ResultAttribute;
+import com.googlecode.jmxtrans.model.Server;
+import com.googlecode.jmxtrans.model.output.support.WriterBasedOutputWriter;
+import com.googlecode.jmxtrans.model.results.CPrecisionValueTransformer;
+import com.googlecode.jmxtrans.model.results.ValueTransformer;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static com.googlecode.jmxtrans.util.NumberUtils.isNumeric;
+
+public class StatsDTelegrafWriter implements WriterBasedOutputWriter {
+
+	private static final Logger log = LoggerFactory.getLogger(StatsDTelegrafWriter.class);
+
+	private final ValueTransformer valueTransformer = new CPrecisionValueTransformer();
+	private final String[] bucketTypes;
+	private final ImmutableMap<String, String> tags;
+	private final ImmutableSet<ResultAttribute> resultAttributesToWriteAsTags;
+
+	public StatsDTelegrafWriter(String bucketType, ImmutableMap<String, String> tags, ImmutableSet<ResultAttribute> resultAttributesToWriteAsTags) {
+		this.bucketTypes = StringUtils.split(bucketType, ",");
+		this.tags = tags;
+		this.resultAttributesToWriteAsTags = resultAttributesToWriteAsTags;
+	}
+
+	@Override
+	public void write(@Nonnull Writer writer, @Nonnull Server server, @Nonnull Query query, @Nonnull Iterable<Result> results) throws IOException {
+
+		int resultIndex = -1;
+		for (Result result : results) {
+			resultIndex++;
+			String bucketType = getBucketType(resultIndex);
+			String attributeName = result.getAttributeName();
+
+			List<String> resultTagList = new ArrayList<>();
+			resultTagList.add(",jmxport=" + server.getPort());
+			//tagList.add("objectName=" + query.getObjectName());
+			resultTagList.add("attribute=" + attributeName);
+
+			for (Map.Entry<String, Object> values : result.getValues().entrySet()) {
+
+				String field = values.getKey();
+				Object value = values.getValue();
+				if (isNotValidValue(value)) {
+					log.debug("Skipping message key[{}] with value: {}.", field, value);
+					continue;
+				}
+				List<String> tagList = new ArrayList(resultTagList);
+
+				boolean isSingleValueAttribute = StringUtils.equals(attributeName, field);
+				if( !isSingleValueAttribute ){
+					tagList.add("resultKey="+values.getKey());
+				}
+
+				for (Map.Entry e : tags.entrySet()) {
+					tagList.add(e.getKey() + "=" + e.getValue());
+				}
+
+				Number actualValue = computeActualValue(value);
+				StringBuilder sb = new StringBuilder(result.getKeyAlias())
+					.append(StringUtils.join(tagList, ","))
+					.append(":").append(actualValue)
+					.append("|").append(bucketType).append("\n");
+
+				String output = sb.toString();
+				if( actualValue.floatValue() < 0 && !StatsDMetricType.GAUGE.getKey().equals(bucketType) )
```

Comment:
I thought so too, but earlier in the loop it checks for a numeric value and continues if not.  I just wrote a test to validate this.